crates/humanode-runtime/src/tests/claims_and_vesting.rs

```diff
@@ -424,7 +424,15 @@ fn direct_claiming_without_vesting_works() {
         // Test preconditions.
         assert!(TokenClaims::claims(ethereum_address).is_some());
         assert_eq!(Balances::free_balance(account_id("Alice")), INIT_BALANCE);
-        assert_eq!(Balances::usable_balance(account_id("Alice")), INIT_BALANCE);
+        // Alice account can't be reaped as the account takes part in consensus (bootnode).
+        // Frozen balance is 0 < existential deposit.
+        //
+        // As a result, usable balance doesn't contain existential deposit.
+        assert_eq!(
+            Balances::usable_balance(account_id("Alice")),
+            INIT_BALANCE
+                - <Balances as frame_support::traits::Currency<AccountId>>::minimum_balance()
+        );
```

Comment:
Not sure this is correct

Reply:
Don't remember exactly the root cause. But it's connected with such changes - https://github.com/paritytech/substrate/commit/f8a1dd4c836d7d29382ba66fc2fce40fc1364546#diff-92414c2ef872b5f2a6af9fde64c13dc4e883a706d3d2da5817caffc6e4d8d0fcR137 containing

```
impl<Balance: Saturating + Copy + Ord> AccountData<Balance> {
	pub fn usable(&self) -> Balance {
		self.free.saturating_sub(self.frozen)
	}

	/// The total balance in this account including any that is reserved and ignoring any frozen.
	pub fn total(&self) -> Balance {
		self.free.saturating_add(self.reserved)
	}
}
```

`Alice` here is bootnode. But need time to remind.

Reply:
Well, this is clearly a change that breaks things - and the question is if this breakage is intended / acceptable; imagine what the consequences of this are.

I think this might be an exact famous case where test found an issue and was "fixed" instead of leading to the actual issue being fixed.
This needs a very serious justification, with proofs in both the intent, the upstream code and upstream test corrections, as well as the implication and cause analysis.